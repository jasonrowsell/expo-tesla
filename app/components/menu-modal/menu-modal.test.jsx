import React from 'react';
import { render } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import { Text } from 'react-native';

import MenuModal from './menu-modal';

describe('<ItemsList />', () => {
  test('has 1 child', () => {
    const onClose = jest.fn();
    const tree = renderer.create(
      <MenuModal onClose={() => onClose()}>
        <Text>
          My Content
        </Text>
      </MenuModal>,
    ).toJSON();
    expect(tree.children.length).toBe(1);
  });

  test('renders correctly', () => {
    const onClose = jest.fn();
    const tree = render(
      <MenuModal onClose={() => onClose()}>
        <Text>
          My Content
        </Text>
      </MenuModal>,
    );
    expect(tree).toMatchSnapshot();
  });
});
