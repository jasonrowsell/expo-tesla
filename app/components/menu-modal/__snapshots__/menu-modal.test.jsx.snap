// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemsList /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(0, 0, 0, 0.5)",
      "flex": 1,
      "height": 1334,
      "left": 0,
      "opacity": 0,
      "position": "absolute",
      "top": -50,
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "display": "flex",
        "elevation": 2,
        "flex": 1,
        "height": "100%",
        "marginLeft": "auto",
        "marginRight": 0,
        "shadowColor": "#000",
        "shadowOpacity": 0.5,
        "shadowRadius": 5,
        "transform": Array [
          Object {
            "translateX": 150,
          },
        ],
        "width": 310,
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "paddingHorizontal": 20,
          "top": 50,
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 30,
            "opacity": 1,
            "width": 30,
          }
        }
        testID="close-icon"
      >
        <Image
          source={1}
          style={
            Object {
              "height": 20,
              "resizeMode": "contain",
              "width": 20,
            }
          }
        />
      </View>
    </View>
    <Text>
      My Content
    </Text>
  </View>
</View>
`;
